src/main/kotlin/org/jetbrains/research/testspark/helpers/psi/kotlin/KotlinPsiMethodWrapper.kt

```diff
@@ -0,0 +1,108 @@
+package org.jetbrains.research.testspark.helpers.psi.kotlin
+
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.parentOfType
+import org.jetbrains.kotlin.idea.refactoring.isInterfaceClass
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.KtPrimaryConstructor
+import org.jetbrains.kotlin.psi.KtSecondaryConstructor
+import org.jetbrains.kotlin.psi.KtTypeReference
+import org.jetbrains.kotlin.psi.psiUtil.containingClassOrObject
+import org.jetbrains.research.testspark.helpers.psi.PsiClassWrapper
+import org.jetbrains.research.testspark.helpers.psi.PsiMethodWrapper
+
+class KotlinPsiMethodWrapper(val psiFunction: KtFunction) : PsiMethodWrapper {
+
+    override val name: String get() = psiFunction.name ?: ""
+
+    override val text: String? = psiFunction.text
+
+    override val containingClass: PsiClassWrapper? = psiFunction.run {
+        parentOfType<KtClass>()?.let { KotlinPsiClassWrapper(it) }
+    }
+
+    override val containingFile: PsiFile = psiFunction.containingFile
+
+    override val methodDescriptor: String
+        get() = psiFunction.run {
+            val parameterTypes = valueParameters.joinToString("") { generateFieldType(it.typeReference) }
+            val returnType = generateReturnDescriptor(psiFunction)
+            return "$name($parameterTypes)$returnType"
+        }
+
+    override val signature: String
+        get() = psiFunction.run {
+            val bodyStart = bodyExpression?.startOffsetInParent ?: textLength
+            text.substring(0, bodyStart).replace('\n', ' ').trim()
+        }
```

Comment:
Is it possible to replace with an empty char `.replace('\n', '').trim()`?

Reply:
why do we want to do this?

Reply:
Just a personal preference.